### PR TITLE
[Rust] Support for withAWSV4Signature option

### PR DESCRIPTION
**Pull request by jerome-jutteau**
Hi @farcaller 

This PR is about supporting AWSV4Signature in rust generator (see issue #11193).

Few notes:
- Option is by default set to `false`
- Only reqwest is supported for now (non-async)
- Created a new petstore sample folder with it's pom.xml and seems to be OK.
- Put in a separate commit other `samples/client/petstore/rust/*` which seems to have changed
- PR is based on master

### PR checklist
 
- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [X] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/rust/reqwest/api.mustache [hunk 1]
```diff
@@ -141,6 +141,27 @@ pub {{#supportAsync}}async {{/supportAsync}}fn {{{operationId}}}(configuration:
...
     {{/hasAuthMethods}}
+    {{#hasAuthMethods}}
+    {{#withAWSV4Signature}}
+    if let Some(ref local_var_aws_v4_key) = local_var_configuration.aws_v4_key {
+        let local_var_new_headers = local_var_aws_v4_key.sign(
+	    &local_var_uri_str,
+	    "{{{httpMethod}}}",
+	    {{#hasBodyParam}}
+	    {{#bodyParams}}
+	    &serde_json::to_string(&{{{paramName}}}).unwrap(),
```
Comment (farcaller) [inline modules/openapi-generator/src/main/resources/rust/reqwest/api.mustache:152]: While this is generally a [safe unwrap](https://docs.serde.rs/serde_json/fn.to_string.html) I'd think that an `expect` with some context would have been more useful.

File: modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache [hunk 2]
```diff
@@ -22,6 +29,44 @@ pub struct ApiKey {
     pub key: String,
 }
 
+{{#withAWSV4Signature}}
+#[derive(Debug, Clone)]
+pub struct AWSv4Key {
+    pub access_key: String,
+    pub secret_key: String,
```
Comment (farcaller) [inline modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache:36]: shouldn't this be wrapped in something like [secrecy](https://crates.io/crates/secrecy)?

File: modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache [hunk 3]
```diff
@@ -22,6 +29,44 @@ pub struct ApiKey {
...
+	    .access_key(self.access_key.as_str())
+	    .secret_key(self.secret_key.as_str())
+	    .region(self.region.as_str())
+	    .service_name(self.service.as_str())
+	    .time(SystemTime::now())
+	    .settings(signing_settings)
+	    .build()
+	    .unwrap();
+	let signable_request = SignableRequest::from(&request);
+	let (mut signing_instructions, _signature) = sign(signable_request, &signing_params).unwrap().into_parts();
```
Comment (farcaller) [inline modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache:58]: I'm a bit concerned about all the `unwrap`s in here. Maybe propagate them up the stack proper instead?

**Review by farcaller — CHANGES_REQUESTED** [review]

File: modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache (hunk 2 above)
Comment (jerome-jutteau) [reply modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache:36]: Didn't knew about secrecy, I will fix this

Comment (farcaller) [reply modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache:36]: FWIW I don't insist on secrecy specifically (just happens to be the one I've used).

**Comment by jerome-jutteau** [comment]
Hi @farcaller, thanks for your review. I pushed a new version which should fix what you pointed out.

**Review by farcaller — APPROVED** [review]
Thanks for your updates, this looks solid.

**Comment by farcaller** [comment]
@wing328 what's with the failing node2 test?

**Comment by wing328** [comment]
@farcaller  please kindly ignore the circleci node 2 failure.

**Comment by wing328** [comment]
Tested the new samples (option) locally and the result is good:
```
   Compiling petstore-reqwest-awsv4signature v1.0.0 (/Users/williamcheng/Code/openapi-generator/samples/client/petstore/rust/reqwest/petstore-awsv4signature)
    Finished dev [unoptimized + debuginfo] target(s) in 6m 02s
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  06:04 min
[INFO] Finished at: 2022-03-02T01:15:49+08:00
[INFO] ------------------------------------------------------------------------

```


Other Rust tests passed via https://app.travis-ci.com/github/OpenAPITools/openapi-generator/builds/247229008

**Comment by wing328** [comment]
FYI. I've pushed 8c023fae198 to replace tabs with 4-space.

**Pull request merged**